src/module.c

```diff
@@ -92,16 +92,38 @@ int module_load(char *module_name, char *driver) {
  * @return 1 if the driver is succesfully unloaded, 0 otherwise
  */
 int module_unload(char *driver) {
-  if (module_is_loaded(driver) == 1) {
+
+  if (module_is_loaded("nvidia_uvm") == 1) {
+    int retries = 30;
+    bb_log(LOG_INFO, "Unloading nvidia_uvm driver\n");
+    char *mod_argv[] = {
+      "modprobe",
+      "-r",
+      "nvidia_uvm",
+      "nvidia_modeset",
```

Comment:
Just came across this line of code. You need to add `driver` as well, otherwise you will unload only `modeset` module, but not `nvidia`. Alternatively, you can change the logic removing `else if` statements and unload modules one by one.